color/src/color.rs

```diff
@@ -179,9 +179,40 @@ impl<CS: ColorSpace> OpaqueColor<CS> {
     ///
     /// This can be useful for choosing contrasting colors, and follows the
     /// WCAG 2.1 spec.
+    #[must_use]
     pub fn relative_luminance(self) -> f32 {
-        let rgb = CS::to_linear_srgb(self.components);
-        0.2126 * rgb[0] + 0.7152 * rgb[1] + 0.0722 * rgb[2]
+        let [r, g, b] = CS::to_linear_srgb(self.components);
+        0.2126 * r + 0.7152 * g + 0.0722 * b
+    }
+
+    /// Map components.
+    #[must_use]
+    pub fn map(self, f: impl Fn(f32, f32, f32) -> [f32; 3]) -> Self {
+        let [x, y, z] = self.components;
+        Self::new(f(x, y, z))
+    }
+
+    /// Map components in a given color space.
+    #[must_use]
+    pub fn map_in<TargetCS: ColorSpace>(self, f: impl Fn(f32, f32, f32) -> [f32; 3]) -> Self {
+        self.convert::<TargetCS>().map(f).convert()
+    }
+
+    /// Map the lightness of the color.
+    ///
+    /// In a color space that naturally has a lightness component, map that value.
+    /// Otherwise, do the mapping in Oklab. The lightness range is normalized so
+    /// that 1.0 is white.
+    #[must_use]
+    pub fn map_lightness(self, f: impl Fn(f32) -> f32) -> Self {
```

Comment:
Another candidate would be `map_hue` due to its natural representation in all polar spaces.